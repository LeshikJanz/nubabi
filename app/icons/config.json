{
  "IcoMoonType": "selection",
  "icons": [
    {
      "icon": {
        "paths": [
          "M66.87 963.766c-11.646 0-23.452-3.052-34.187-9.45-31.751-18.914-42.165-59.981-23.238-91.745l163.078-273.764c11.512-19.329 31.992-31.577 54.466-32.608 22.622-1.044 43.972 9.343 57.197 27.561l69.632 95.949 147.885-251.41c11.351-19.316 31.644-31.671 54.025-32.902 22.381-1.098 43.892 8.848 57.304 26.812l128.971 172.822 338.417-562.6c19.075-31.657 60.169-41.897 91.853-22.849 31.671 19.048 41.897 60.169 22.849 91.839l-389.791 648.027c-11.498 19.101-31.751 31.242-54.038 32.34-22.207 1.218-43.611-8.942-56.956-26.812l-128.342-171.979-148.621 252.681c-11.445 19.463-31.925 31.831-54.466 32.915-22.622 1.044-44.119-9.263-57.384-27.534l-70.007-96.457-111.088 186.476c-12.516 21.016-34.749 32.688-57.558 32.688z"
        ],
        "attrs": [],
        "width": 1205,
        "isMulticolor": false,
        "isMulticolor2": false,
        "grid": 0,
        "tags": [
          "growth"
        ],
        "colorPermutations": {}
      },
      "attrs": [],
      "properties": {
        "order": 20,
        "id": 10,
        "name": "growth",
        "prevSize": 32,
        "code": 59654
      },
      "setIdx": 1,
      "setId": 0,
      "iconIdx": 0
    },
    {
      "icon": {
        "paths": [
          "M625.778 176.655h-113.778v718.596h113.778z",
          "M113.778 92.34c0 37.805-17.93 44.916-32.438 45.391 1.677-0.055 2.958-0.058 7.132-0.025 2.035 0.019 2.035 0.019 4.323 0.026h311.392c53.678 0 113.049 31.434 136.923 71.091 16.204 26.918 51.162 35.603 78.079 19.398s35.603-51.162 19.398-78.079c-45.098-74.913-141.942-126.188-234.4-126.188h-311.392c-1.396-0.004-1.396-0.004-3.31-0.022-5.054-0.047-8.12-0.041-11.864 0.082-12.687 0.415-23.333 1.941-34.644 6.426-24.55 9.736-42.977 31.723-42.977 61.9v755.005c0 31.419 25.47 56.889 56.889 56.889s56.889-25.47 56.889-56.889v-755.005z",
          "M1167.719 92.34c0.051-3.376 0.076-7.12-0.094-10.296-0.816-15.231-4.69-28.495-18.966-41.849-13.511-12.638-25.623-15.66-40.173-16.221-2.79-0.108-6.371-0.067-9.657-0.020h-311.392c-94.645 0-202.031 52.013-246.424 126.298-16.117 26.97-7.319 61.899 19.651 78.016s61.899 7.319 78.016-19.651c22.26-37.249 91.707-70.885 148.757-70.885h311.392c5.743-0.043 5.818-0.044 5.27-0.065-10.838-0.418-20.931-2.936-33.165-14.38-18.422-17.232-16.993 9.446-16.993-30.947v755.005c0 31.419 25.47 56.889 56.889 56.889s56.889-25.47 56.889-56.889v-755.005z",
          "M414.654 928.187c54.009 0 110.547 25.044 135.837 58.426 18.973 25.044 54.655 29.965 79.699 10.992s29.965-54.655 10.992-79.699c-48.018-63.382-138.58-103.498-226.527-103.498h-333.812c-31.419 0-56.889 25.47-56.889 56.889s25.47 56.889 56.889 56.889h333.812z",
          "M777.018 814.409c-87.948 0-178.51 40.116-226.527 103.498-18.973 25.044-14.051 60.726 10.992 79.699s60.726 14.051 79.699-10.992c25.29-33.382 81.828-58.426 135.837-58.426h333.812c31.419 0 56.889-25.47 56.889-56.889s-25.47-56.889-56.889-56.889h-333.812z",
          "M928.212 395.303l53.97-17.99-35.98-107.939-278.606 92.869 35.98 107.939 53.97-17.99z",
          "M928.212 622.858l53.97-17.99-35.98-107.939-278.606 92.869 35.98 107.939 53.97-17.99z"
        ],
        "attrs": [],
        "width": 1195,
        "isMulticolor": false,
        "isMulticolor2": false,
        "grid": 0,
        "tags": [
          "library"
        ],
        "colorPermutations": {
          "111613014811581711881": [
            {
              "f": 2
            },
            {
              "f": 2
            },
            {
              "f": 2
            },
            {
              "f": 2
            },
            {
              "f": 2
            },
            {
              "f": 2
            },
            {
              "f": 2
            }
          ]
        }
      },
      "attrs": [],
      "properties": {
        "order": 19,
        "id": 9,
        "name": "library",
        "prevSize": 32,
        "code": 59655
      },
      "setIdx": 1,
      "setId": 0,
      "iconIdx": 1
    },
    {
      "icon": {
        "paths": [
          "M799.363 918.187v-669.004c0-30.173-19.612-54.613-43.801-54.613h-711.762c-24.2 0-43.801 24.439-43.801 54.613v669.004c0 30.16 19.601 54.613 43.801 54.613h711.762c24.189 0 43.801-24.453 43.801-54.613zM76.651 665.787v-384.744h646.061v518.815h-646.061v-134.071zM899.864 671.935c-3.219 15.316-16.885 25.844-32.128 25.844-2.212 0-4.468-0.216-6.712-0.681-17.761-3.643-29.182-20.807-25.492-38.339l97.413-463.053-627.96-125.189-15.516 68.993c-3.942 17.478-21.484 28.535-39.191 24.623-17.717-3.891-28.887-21.207-24.955-38.685l22.503-100.047c3.876-17.24 21.035-28.254 38.578-24.752l691.92 137.954c8.574 1.697 16.097 6.712 20.915 13.922 4.818 7.199 6.526 15.997 4.752 24.45l-104.125 494.961z",
          "M256.93 525.541l139.948 164.116c2.412 2.83 6.357 2.792 8.738 0l139.94-164.116-5.579 7.017c8.235-10.11 14.137-22.294 16.832-35.689 1.077-5.359 1.65-10.912 1.65-16.595 0-44.997-35.512-81.48-79.321-81.48-36.14 0-66.63 24.831-76.194 58.785l-3.3-0.062c-9.571-33.961-40.061-58.785-76.202-58.785-43.802 0-79.314 36.475-79.314 81.48 0 5.683 0.566 11.228 1.642 16.587 2.695 13.395 8.597 25.587 16.84 35.697z",
          "M388.5 696.618c6.786 7.96 18.735 7.924 25.492 0.001l139.941-164.118-8.378-6.96-8.614-6.673-5.579 7.017c-8.961 11.27 8.053 24.612 17.148 13.446 9.476-11.633 16.041-25.458 19.039-40.356 1.232-6.132 1.86-12.389 1.86-18.701 0-50.874-40.316-92.289-90.272-92.289-40.56 0-75.719 27.556-86.744 66.698l10.55 2.895 0.207-10.807-3.3-0.062-0.207 10.807 10.55-2.897c-11.033-39.147-46.189-66.697-86.751-66.697-49.952 0-90.264 41.411-90.264 92.289 0 6.327 0.624 12.577 1.851 18.69 3 14.91 9.57 28.743 19.051 40.371 3.792 4.65 10.685 5.386 15.396 1.643s5.456-10.547 1.665-15.198c-7.273-8.92-12.321-19.55-14.63-31.025-0.95-4.727-1.433-9.568-1.433-14.482 0-39.127 30.707-70.671 68.363-70.671 30.531 0 57.206 20.903 65.652 50.873 1.297 4.603 5.503 7.82 10.342 7.91l3.3 0.062c4.994 0.093 9.419-3.161 10.757-7.912 8.44-29.964 35.115-50.871 65.644-50.871 37.661 0 68.371 31.548 68.371 70.671 0 4.894-0.487 9.742-1.441 14.493-2.307 11.465-7.353 22.091-14.625 31.018l8.534 6.773 8.614 6.673 5.579-7.017c8.861-11.145-7.75-24.472-16.992-13.634l-139.94 164.116c1.974-2.314 6.034-2.327 8.018 0.001l-139.949-164.118c-3.894-4.567-10.802-5.153-15.429-1.309s-5.22 10.663-1.326 15.23l139.948 164.116z"
        ],
        "attrs": [],
        "isMulticolor": false,
        "isMulticolor2": false,
        "grid": 0,
        "tags": [
          "memories"
        ],
        "colorPermutations": {
          "111613014811581711881": [
            {
              "f": 2
            },
            {
              "f": 2
            },
            {
              "f": 2
            }
          ]
        }
      },
      "attrs": [],
      "properties": {
        "order": 18,
        "id": 8,
        "name": "memories",
        "prevSize": 32,
        "code": 59656
      },
      "setIdx": 1,
      "setId": 0,
      "iconIdx": 2
    },
    {
      "icon": {
        "paths": [
          "M961.86 63.891c-37.634-37.634-87.521-58.639-140.909-58.639s-103.275 21.005-140.909 58.639c-16.629 16.629-29.757 35.884-39.385 56.889l-104.15-104.15c-14.003-14.003-36.759-14.003-49.887 0l-472.615 472.615c-14.003 14.003-14.003 35.884 0 49.887l472.615 472.615c7.002 7.002 15.754 10.503 25.381 10.503s18.379-3.501 25.381-10.503l150.537-150.537c9.627-9.627 13.128-24.506 7.877-37.634s-16.629-21.88-30.632-22.756c-30.632-2.626-59.515-15.754-81.395-36.759-24.506-24.506-37.634-56.889-37.634-91.022s13.128-66.516 37.634-91.022c24.506-24.506 56.889-37.634 91.022-37.634s66.516 13.128 91.022 37.634c21.88 21.88 35.009 50.762 36.759 81.395 0.875 14.003 9.627 26.256 22.756 30.632 13.128 5.251 28.007 1.75 37.634-7.877l146.161-146.161c7.002-7.002 10.503-15.754 10.503-25.381s-3.501-18.379-10.503-25.381l-104.15-104.15c21.005-9.627 40.26-23.631 56.889-39.385 77.894-77.894 77.894-204.8 0-281.819v0zM911.973 294.947c-21.005 21.005-48.137 34.133-77.894 36.759-14.003 1.75-25.381 10.503-29.757 23.631s-1.75 27.132 8.752 36.759l121.655 121.655-77.019 77.019c-9.627-21.88-23.631-42.885-41.135-59.515-37.634-37.634-87.521-58.639-140.909-58.639s-103.275 21.005-140.909 58.639c-37.634 37.634-58.639 87.521-58.639 140.909s21.005 103.275 58.639 140.909c17.504 17.504 37.634 31.508 59.515 41.135l-81.395 81.395-423.603-421.853 421.853-421.853 121.655 121.655c9.627 9.627 24.506 13.128 36.759 8.752 13.128-4.376 21.88-16.629 23.631-29.757 2.626-29.757 15.754-56.889 36.759-77.894 24.506-24.506 56.889-37.634 91.022-37.634s66.516 13.128 91.022 37.634c49.887 49.012 49.887 130.407 0 180.294v0z"
        ],
        "attrs": [],
        "isMulticolor": false,
        "isMulticolor2": false,
        "grid": 0,
        "tags": [
          "stimulation"
        ],
        "colorPermutations": {
          "111613014811581711881": [
            {
              "f": 2
            }
          ]
        }
      },
      "attrs": [],
      "properties": {
        "order": 21,
        "id": 7,
        "name": "stimulation",
        "prevSize": 32,
        "code": 59657
      },
      "setIdx": 1,
      "setId": 0,
      "iconIdx": 3
    },
    {
      "icon": {
        "paths": [
          "M459.625 596.070c-37.475 0-67.844-30.369-67.844-67.844s30.369-67.844 67.844-67.844c37.475 0 67.844 30.369 67.844 67.844s-30.369 67.844-67.844 67.844zM459.625 352.593c-97.006 0-175.633 78.628-175.633 175.633s78.628 175.633 175.633 175.633c97.006 0 175.633-78.628 175.633-175.633s-78.628-175.633-175.633-175.633z",
          "M862.738 518.106c0-42.557-6.893-84.412-20.151-124.69l-51.193 16.85 31.266 43.898 62.614-44.597c33.386-23.741 41.169-70.119 17.357-103.5l-94.142-132.177c-23.845-33.354-70.087-41.118-103.436-17.403l-63.889 44.018 30.581 44.379 32.881-42.702c-37.614-28.963-81.355-51.345-127.923-65.385l-15.558 51.6h53.895v-60.322c0-48.082-26.938-74.185-69.964-74.185v53.895l-162.318-53.895c-12.547 0-24.702 4.378-34.367 12.379l-24.51 20.29c-12.368 10.239-19.528 25.459-19.528 41.516v60.322h53.895l-16.029-51.456c-45.68 14.229-87.722 37.074-127.067 68.096l33.369 42.322 31.204-43.942-67.545-47.965c-33.909-23.782-79.697-15.483-103.123 18.099l-92.784 132.998c-23.48 33.66-15.012 79.8 18.389 103.225l69.43 48.507 30.867-44.18-51.383-16.26c-12.227 38.639-18.52 79.059-18.52 120.265 0 43.096 7.112 85.509 20.755 126.34l51.117-17.079-31.268-43.897-63.122 44.962c-33.383 23.813-41.16 70.128-17.355 103.499l94.166 132.181c23.767 33.348 70.050 41.123 103.412 17.399l62.277-45.882-31.972-43.387-32.689 42.85c40.043 30.548 78.084 51.167 122.503 64.381l15.368-51.657h-53.895v77.872c0 38.853 32.508 88.164 78.405 88.164h162.318c53.98 0 69.964-43.772 69.964-88.164v-77.872h-53.895l15.581 51.593c46.622-14.080 90.513-36.543 128.158-65.552l-32.896-42.69-30.521 44.42 59.206 40.681c33.134 23.189 79.925 15.134 103.469-18.511l93.26-132.974c23.491-33.495 15.58-79.857-18.159-103.392l-56.804-39.834-30.944 44.126 51.208 16.804c13.182-40.17 20.034-81.93 20.034-124.38zM760.552 669.809l56.913 39.91c-15.305-10.676-18.898-31.734-8.312-46.829l-93.292 133.019c10.676-15.256 31.143-18.779 46.233-8.219l-59.584-40.944c-19.275-13.244-44.893-12.546-63.418 1.729-27.224 20.978-59.37 37.43-93.528 47.746-22.749 6.87-38.314 27.83-38.314 51.593v77.872c0-12.258 2.69-19.625 37.825-19.625h-162.318c18.074 0 24.708 5.492 28.534 11.296 0.494 0.75 0.85 1.687 0.85 8.329v-77.872c0-23.846-15.671-44.858-38.527-51.657-30.865-9.182-57.568-23.656-87.861-46.766-19.040-14.525-45.381-14.744-64.661-0.538l-61.539 45.348c15.844-11.276 36.804-7.755 47.577 7.36l-94.179-132.2c10.763 15.088 7.238 36.080-7.849 46.842l63.092-44.941c19.304-13.75 27.359-38.497 19.849-60.976-10.018-29.982-15.199-60.881-15.199-92.181 0-30.129 4.586-59.584 13.497-87.746 7.109-22.466-1.201-46.945-20.517-60.44l-69.35-48.452c15.076 10.573 18.901 31.414 8.203 46.751l92.784-132.998c-10.575 15.16-32.086 19.059-47.431 8.297l67.286 47.783c19.5 13.847 45.792 13.188 64.573-1.62 29.342-23.135 59.767-39.667 92.387-49.828 22.523-7.016 37.866-27.865 37.866-51.456v-60.322h-53.895l34.367 41.516 24.51-20.29-34.367-41.516v53.895h162.318v-53.895c-15.757 53.895-37.825 32.511-37.825 20.29v60.322c0 23.773 15.576 44.738 38.337 51.6 34.088 10.278 66.080 26.648 93.276 47.589 18.546 14.28 44.188 14.958 63.462 1.676l64.542-44.475c-14.441 10.276-35.359 6.764-46.119-8.287l94.109 132.13c-10.765-15.092-7.239-36.107 7.917-46.884l-62.647 44.621c-19.232 13.698-27.309 38.32-19.926 60.748 9.725 29.546 14.747 60.041 14.747 90.99 0 30.881-4.994 61.312-14.661 90.773-7.402 22.557 0.828 47.3 20.265 60.93z"
        ],
        "attrs": [
          {
            "fill": "rgb(0, 0, 0)"
          },
          {
            "fill": "rgb(0, 0, 0)"
          }
        ],
        "width": 970,
        "isMulticolor": false,
        "isMulticolor2": false,
        "grid": 0,
        "tags": [
          "settings"
        ],
        "colorPermutations": {
          "111613014811581711881": [
            {
              "f": 0
            },
            {
              "f": 0
            }
          ]
        }
      },
      "attrs": [
        {
          "fill": "rgb(0, 0, 0)"
        },
        {
          "fill": "rgb(0, 0, 0)"
        }
      ],
      "properties": {
        "order": 27,
        "id": 6,
        "name": "settings",
        "prevSize": 32,
        "code": 59653
      },
      "setIdx": 1,
      "setId": 0,
      "iconIdx": 4
    },
    {
      "icon": {
        "paths": [
          "M426.667 975.644c54.756 0 99.556-44.8 99.556-99.556h-199.111c0 54.756 44.8 99.556 99.556 99.556v0zM725.333 676.978v-248.889c0-152.818-81.138-280.747-224-314.596v-33.849c0-41.316-33.351-74.667-74.667-74.667s-74.667 33.351-74.667 74.667v33.849c-142.364 33.849-224 161.28-224 314.596v248.889l-99.556 99.556v49.778h796.444v-49.778l-99.556-99.556zM625.778 726.756h-398.222v-298.667c0-123.449 75.164-224 199.111-224s199.111 100.551 199.111 224v298.667z"
        ],
        "attrs": [
          {
            "fill": "rgb(0, 0, 0)"
          }
        ],
        "width": 853,
        "isMulticolor": false,
        "isMulticolor2": false,
        "grid": 0,
        "tags": [
          "alerts"
        ],
        "colorPermutations": {
          "111613014811581711881": [
            {
              "f": 0
            }
          ]
        }
      },
      "attrs": [
        {
          "fill": "rgb(0, 0, 0)"
        }
      ],
      "properties": {
        "order": 26,
        "id": 5,
        "name": "alerts",
        "prevSize": 32,
        "code": 59648
      },
      "setIdx": 1,
      "setId": 0,
      "iconIdx": 5
    },
    {
      "icon": {
        "paths": [
          "M807.701 410.695c0 37.803-30.646 68.449-68.449 68.449s-68.449-30.646-68.449-68.449c0-37.803 30.646-68.449 68.449-68.449s68.449 30.646 68.449 68.449z",
          "M533.904 410.695c0 37.803-30.646 68.449-68.449 68.449s-68.449-30.646-68.449-68.449c0-37.803 30.646-68.449 68.449-68.449s68.449 30.646 68.449 68.449z",
          "M1201.42 528.975c2.19-11.499 3.286-23.547 3.286-36.141s-1.095-24.642-3.286-36.141c-13.69-82.687-74.473-150.041-153.874-173.587-29.022-61.33-70.092-114.995-119.923-159.35-86.52-77.211-200.419-123.756-325.271-123.756s-238.751 46.545-325.271 123.756c-50.379 44.355-91.448 98.567-119.923 159.35-79.401 23.547-140.184 90.353-153.874 173.587-2.19 11.499-3.286 23.547-3.286 36.141s1.095 24.642 3.286 36.141c13.69 82.687 74.473 150.041 153.874 173.587 28.475 60.783 69.544 114.447 118.828 158.255 86.52 77.758 200.967 124.851 326.366 124.851s239.846-47.093 326.913-124.851c49.283-43.807 90.353-98.019 118.828-158.255 78.853-23.547 139.636-90.353 153.326-173.587v0zM985.668 602.353c-5.476 0-10.404-1.095-15.88-1.643-10.952 36.689-26.832 70.64-47.093 101.852-68.449 104.59-186.182 173.587-320.342 173.587s-251.893-68.997-320.342-173.587c-20.261-31.213-36.141-65.164-47.093-101.852-5.476 0.548-10.404 1.643-15.88 1.643-60.235 0-109.519-49.283-109.519-109.519s49.283-109.519 109.519-109.519c5.476 0 10.404 1.095 15.88 1.643 10.952-36.689 26.832-70.64 47.093-101.852 68.449-104.59 186.182-173.587 320.342-173.587s251.893 68.997 320.342 173.587c20.261 31.213 36.141 65.164 47.093 101.852 5.476-0.548 10.404-1.643 15.88-1.643 60.235 0 109.519 49.283 109.519 109.519s-49.283 109.519-109.519 109.519v0zM355.936 602.353c41.617 96.924 136.351 164.278 246.417 164.278s204.8-67.354 246.417-164.278h-492.834z"
        ],
        "attrs": [],
        "width": 1205,
        "isMulticolor": false,
        "isMulticolor2": false,
        "grid": 0,
        "tags": [
          "changeBabyIcon"
        ],
        "colorPermutations": {
          "111": [
            {
              "f": 0
            },
            {
              "f": 0
            },
            {
              "f": 0
            }
          ],
          "111613014811581711881": []
        }
      },
      "attrs": [],
      "properties": {
        "order": 25,
        "id": 4,
        "name": "changeBabyIcon",
        "prevSize": 32,
        "code": 59649
      },
      "setIdx": 1,
      "setId": 0,
      "iconIdx": 6
    },
    {
      "icon": {
        "paths": [
          "M1000.068 350.322c31.909-31.909 31.909-83.674 0-115.583l-210.803-210.803c-31.982-31.901-83.682-31.901-115.583 0l-579.713 579.713c-8.252 8.252-13.767 18.843-15.796 30.336l-52.719 298.627c-6.866 38.892 27.015 72.776 65.907 65.914l298.651-52.694c11.495-2.028 22.088-7.543 30.342-15.797l579.713-579.713zM339.902 849.582l40.227 40.227-9.885-56.024-298.651 52.694 65.907 65.914 52.719-298.627-56.023-9.89 40.227 40.227 579.713-579.713c-12.515 12.515-32.776 12.515-45.272 0.051l210.752 210.752c-12.524-12.524-12.524-32.8 0-45.323l-579.713 579.713z",
          "M587.501 190.571c-22.216-22.216-58.237-22.216-80.453 0s-22.216 58.237 0 80.453l245.932 245.932c22.216 22.216 58.237 22.216 80.453 0s22.216-58.237 0-80.453l-245.932-245.932z"
        ],
        "attrs": [
          {
            "fill": "rgb(0, 0, 0)"
          },
          {
            "fill": "rgb(0, 0, 0)"
          }
        ],
        "isMulticolor": false,
        "isMulticolor2": false,
        "grid": 0,
        "tags": [
          "editProfile"
        ],
        "colorPermutations": {
          "111": [
            {
              "f": 2
            },
            {
              "f": 2
            }
          ],
          "111613014811581711881": [
            {
              "f": 0
            },
            {
              "f": 0
            }
          ]
        }
      },
      "attrs": [
        {
          "fill": "rgb(0, 0, 0)"
        },
        {
          "fill": "rgb(0, 0, 0)"
        }
      ],
      "properties": {
        "order": 24,
        "id": 3,
        "name": "editProfile",
        "prevSize": 32,
        "code": 59650
      },
      "setIdx": 1,
      "setId": 0,
      "iconIdx": 7
    },
    {
      "icon": {
        "paths": [
          "M333.53 987.373c31.649-0.131 56.827-25.31 56.969-56.969l1.415-511.895 114.621 114.621c22.297 22.297 60.053 22.474 82.526 0.002l0.295-0.295c22.472-22.472 22.321-59.348 0.025-81.644l-212.24-212.424c-10.704-10.704-24.763-16.259-39.96-16.195-15.207 0.053-29.305 6.585-40.099 17.379l-213.776 213.776c-22.472 22.472-22.633 58.751-0.336 81.048 22.307 22.307 58.576 22.136 81.048-0.336l115.535-115.535-2.56 511.931c-0.121 31.659 24.879 56.66 56.538 56.538z",
          "M633.6 124.121c34.561 0 62.578-27.785 62.578-62.061s-28.017-62.061-62.578-62.061h-571.022c-34.561 0-62.578 27.785-62.578 62.061s28.017 62.061 62.578 62.061h571.022z"
        ],
        "attrs": [
          {
            "fill": "rgb(0, 0, 0)"
          },
          {
            "fill": "rgb(0, 0, 0)"
          }
        ],
        "width": 701,
        "isMulticolor": false,
        "isMulticolor2": false,
        "grid": 0,
        "tags": [
          "height"
        ],
        "colorPermutations": {
          "111": [
            {
              "f": 1
            },
            {
              "f": 1
            }
          ],
          "111613014811581711881": [
            {
              "f": 0
            },
            {
              "f": 0
            }
          ]
        }
      },
      "attrs": [
        {
          "fill": "rgb(0, 0, 0)"
        },
        {
          "fill": "rgb(0, 0, 0)"
        }
      ],
      "properties": {
        "order": 23,
        "id": 2,
        "name": "height",
        "prevSize": 32,
        "code": 59651
      },
      "setIdx": 1,
      "setId": 0,
      "iconIdx": 8
    },
    {
      "icon": {
        "paths": [
          "M886.163 62.054c1.786-15.802-3.497-31.606-14.523-43.421-11.083-11.839-26.873-18.633-43.325-18.633h-768.588c-16.536 0-32.325 6.794-43.336 18.599-11.013 11.814-16.294 27.616-14.51 43.421 13.035 115.137 74.29 219.686 169.332 290.71-78.874 70.36-128.244 170.826-128.244 282.385 0 91.595 33.279 175.707 88.826 241.827l-56.342 60.232c-19.634 20.992-17.835 53.297 4.017 72.16 10.145 8.78 22.854 13.103 35.529 13.103 14.561 0 29.071-5.719 39.581-16.934l54.404-58.154c66.047 45.974 147.238 73.095 235.038 73.095s168.991-27.119 235.038-73.095l54.406 58.156c10.509 11.215 25.019 16.934 39.581 16.934 12.656 0 25.382-4.322 35.527-13.103 21.852-18.864 23.651-51.168 4.017-72.16l-56.344-60.232c55.545-66.12 88.826-150.232 88.826-241.827 0-111.559-49.37-212.025-128.244-282.385 95.045-71.026 156.302-175.572 169.334-290.678v0zM444.022 897.418c-150.777 0-272.998-117.409-272.998-262.303 0-144.852 122.222-262.265 272.998-262.265 150.775 0 272.994 117.413 272.994 262.265 0.002 144.894-122.219 262.303-272.994 262.303v0z",
          "M463.78 585.112l-48.672-137.569c-6.303-17.797-26.457-27.314-44.983-21.286-18.542 6.087-28.465 25.414-22.162 43.247l48.67 137.533c-5.263 8.181-8.45 17.763-8.45 28.077 0 29.645 25.003 53.665 55.84 53.665s55.84-24.022 55.84-53.665c0-22.892-15.030-42.316-36.082-50.003v0z"
        ],
        "attrs": [
          {
            "fill": "rgb(0, 0, 0)"
          },
          {
            "fill": "rgb(0, 0, 0)"
          }
        ],
        "width": 896,
        "isMulticolor": false,
        "isMulticolor2": false,
        "grid": 0,
        "tags": [
          "weight"
        ],
        "colorPermutations": {
          "111": [
            {
              "f": 1
            },
            {
              "f": 1
            }
          ],
          "111613014811581711881": [
            {
              "f": 0
            },
            {
              "f": 0
            }
          ]
        }
      },
      "attrs": [
        {
          "fill": "rgb(0, 0, 0)"
        },
        {
          "fill": "rgb(0, 0, 0)"
        }
      ],
      "properties": {
        "order": 22,
        "id": 0,
        "name": "weight",
        "prevSize": 32,
        "code": 59652
      },
      "setIdx": 1,
      "setId": 0,
      "iconIdx": 9
    }
  ],
  "height": 1024,
  "metadata": {
    "name": "nubabi"
  },
  "preferences": {
    "showGlyphs": true,
    "showQuickUse": true,
    "showQuickUse2": false,
    "showSVGs": false,
    "fontPref": {
      "prefix": "icon-",
      "metadata": {
        "fontFamily": "nubabi",
        "majorVersion": 1,
        "minorVersion": 0
      },
      "metrics": {
        "emSize": 1024,
        "baseline": 6.25,
        "whitespace": 50
      },
      "embed": false
    },
    "imagePref": {
      "prefix": "icon-",
      "png": true,
      "useClassSelector": true,
      "color": 0,
      "bgColor": 16777215,
      "classSelector": ".icon"
    },
    "historySize": 100,
    "showCodes": true,
    "gridSize": 16
  }
}