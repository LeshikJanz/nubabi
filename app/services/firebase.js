import * as firebase from 'firebase';

const api = {
  async login(email, pass) {
    try {
      const response = await firebase.auth()
        .signInWithEmailAndPassword(email, pass);
      return response;
    } catch (error) {
      console.log(error.toString());
      return error;
    }
  },
  async logout() {
    try {
      return firebase.auth().signOut();
    } catch (error) {
      console.log(error.toString());
      return error;
    }
  },
};

export default api;
