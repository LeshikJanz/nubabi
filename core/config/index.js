// @flow

const config = {
  appName: process.env.NUBABI_APP_NAME || 'app',
  appVersion: process.env.NUBABI_APP_VERSION,
  apiUrl: process.env.NUBABI_API_URL,
  environment: process.env.NUBABI_ENV || 'development',
  graphqlEndpoint: process.env.NUBABI_GRAPHQL_ENDPOINT,
  firebase: {
    apiKey: process.env.NUBABI_FIREBASE_API_KEY,
    authDomain: process.env.NUBABI_FIREBASE_AUTH_DOMAIN,
    databaseURL: process.env.NUBABI_FIREBASE_DATABASE_URL,
    storageBucket: process.env.NUBABI_FIREBASE_STORAGE_BUCKET,
    messagingSenderId: process.env.NUBABI_FIREBASE_MESSAGING_SENDER_ID,
  },
  sentryUrl: process.env.NUBABI_SENTRY_URL,
};

export default config;
