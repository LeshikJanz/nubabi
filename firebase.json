{
  "database": {
    "rules": "platforms/firebase/rules.bolt"
  },
  "functions": {
    "source": "platforms/firebase/functions"
  },
  "hosting": {
    "public": "platforms/web/build",
    "ignore": ["firebase.json", "**/.*", "**/node_modules/**"]
  }
}
