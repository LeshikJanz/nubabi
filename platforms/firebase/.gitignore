.firebaserc
/functionsES6/node_modules
.vscode/*
!.vscode/settings.json
!.vscode/tasks.json
!.vscode/launch.json
!.vscode/extensions.json
rules.json
/functions
/node_modules