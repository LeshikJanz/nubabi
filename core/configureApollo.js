// @flow
import type { Action, State } from './types';
import { ApolloClient } from 'apollo-client';
// $FlowFixMe$
import { IntrospectionFragmentMatcher } from 'react-apollo';
import config from './config';
import ServerNetworkInterface from './helpers/serverNetworkInterface';

let client = null; // singleton

const networkInterface = new ServerNetworkInterface(config.graphqlEndpoint);

export const configureApollo = (): ApolloClient => {
  if (!client) {
    client = new ApolloClient({
      networkInterface,
      dataIdFromObject: obj => obj.id,
      fragmentMatcher: new IntrospectionFragmentMatcher({
        introspectionQueryResultData: require('../platforms/server/graphql/introspection.json')
          .data,
      }),
    });
  }

  return client;
};

export const authTokenMiddleware = (store) => ({
  applyMiddleware(req, next) {
    if (!req.options.headers) {
      req.options.headers = {};
    }

    const { token } = store.getState().auth;
    req.options.headers.authorization = token ? `Bearer ${token}` : null;
    next();
  },
});

export const configureApolloAuth = (store: Store<Action, State>) => {
  networkInterface.use([authTokenMiddleware(store)]);
};
