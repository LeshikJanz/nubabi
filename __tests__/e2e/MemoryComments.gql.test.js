/* eslint-disable graphql/template-strings */
// This file was autogenerated
import fetch from "isomorphic-fetch";
import fs from "fs";
import path from "path";
import nock from "nock";
import gql from "graphql-tag";
import firebase from "firebase";
import { ApolloClient } from "apollo-client";
import { IntrospectionFragmentMatcher } from "react-apollo";
import config from "core/config";
import { authTokenMiddleware } from "core/configureApollo";
import ServerNetworkInterface from "../../core/helpers/serverNetworkInterface";

const networkInterface = new ServerNetworkInterface(
  process.env.NUBABI_GRAPHQL_ENDPOINT
);
const apollo = new ApolloClient({
  networkInterface,
  fragmentMatcher: new IntrospectionFragmentMatcher({
    introspectionQueryResultData: require("../../platforms/server/graphql/introspection.json")
      .data
  })
});

const start = async () => {
  firebase.initializeApp(config.firebase);
  const user = await firebase
    .auth()
    .signInWithEmailAndPassword(
      process.env.NUBABI_E2E_USER_NAME,
      process.env.NUBABI_E2E_USER_PASSWORD
    );

  const token = await user.getToken();

  networkInterface.use([
    authTokenMiddleware({
      getState: () => ({ auth: { token } })
    })
  ]);

  return true;
};

beforeAll(() => start());

describe("E2E GraphQL", () => {
  test("MemoryComments GQL", done => {
    const queryString = fs.readFileSync(
      path.resolve(__dirname, "../../__generated__/MemoryComments.graphql"),
      "utf-8"
    );

    const query = gql`${queryString}`;
    const variables = {
      memoryId: "TWVtb3J5Oi1MMEF0c2U2Z2lXRlAwcmJlNXZy",
      babyId: process.env.NUBABI_E2E_BABY_ID,
    };

    apollo
      .query({ query, variables })
      .then(resp => {
        expect(JSON.parse(JSON.stringify(resp.data))).toMatchSnapshot(
          "MemoryComments"
        );
        done();
      })
      .catch(err => done(err));
  });
});
