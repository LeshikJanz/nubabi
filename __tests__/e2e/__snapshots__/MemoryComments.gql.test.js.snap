// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemoryComments 1`] = `
Object {
  "viewer": Object {
    "__typename": "Viewer",
    "baby": Object {
      "__typename": "Baby",
      "id": "QmFieTotS2xOdXRVeUFEbU5QQTlBQkh0Yw==",
      "memory": Object {
        "__typename": "Memory",
        "comments": Object {
          "__typename": "CommentConnection",
          "count": 0,
          "edges": Array [],
        },
        "id": "TWVtb3J5Oi1MMEF0c2U2Z2lXRlAwcmJlNXZy",
      },
    },
  },
}
`;
